test/integration/framework/helm_utils.go

```
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archiver"
+
+	"k8s.io/helm/pkg/helm/environment"
+
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/helmpath"
+	"k8s.io/helm/pkg/repo"
+)
+
+const (
+	stableRepository = "stable"
+)
+
+//TODO call only if chart is not downloaded
+//downloadChart downloads a native chart with <name> to <downloadDestination> from <stableRepoURL>
+func downloadChart(ctx context.Context, name, downloadDestination, stableRepoURL string, helmSettings HelmAccess) (string, error) {
+	providers := getter.All(environment.EnvSettings{})
+	dl := downloader.ChartDownloader{
+		Getters:  providers,
+		HelmHome: helmpath.Home(helmSettings.HelmPath),
+		Out:      os.Stdout,
+	}
+
+	err := ensureCacheIndex(ctx, helmSettings, stableRepoURL, providers)
+	if err != nil {
+		return "", err
+	}
+
+	// Download the chart
+	filename, _, err := dl.DownloadTo(name, "", downloadDestination)
+	if err != nil {
+		return "", err
+	}
+
+	lname, err := filepath.Abs(filename)
+	if err != nil {
+		return "", err
+	}
+
+	_ = archiver.Unarchive(lname, downloadDestination)
```

This error is not handled